metricbeat/helper/elastic/json.go

```diff
@@ -0,0 +1,36 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package elastic
+
+import (
+	"bytes"
+	"encoding/json"
+)
+
+// Unmarshal has the same signature as json.Unmarshal, except it uses
+// json.(*Decoder).UseNumber to unmarshal numbers into json.Number instead
+// of float64. This prevents the issue of sending a number in scientific
+// notation, unquoted, to Elasticsearch. Note that, by default, Elasticsearch
+// is able to handle numbers that are quoted as long as the mapping is of a
+// numeric type, so sending all unmarshalled numbers — in scientific notation
+// or otherwise — as quoted strings to Elasticsearch is okay!
+func Unmarshal(data []byte, v interface{}) error {
+	dec := json.NewDecoder(bytes.NewReader(data))
+	dec.UseNumber()
+	return dec.Decode(v)
```

Comment:
Do we also need to run

https://github.com/elastic/beats/blob/35a7882f69d6daf9ab72f36c6509a00aaa2904d0/libbeat/common/jsontransform/transform.go#L29

on the output to convert the `json.Number` back to a basic type?

Reply:
I decided to go a different way altogether with the solution here. See https://github.com/elastic/beats/pull/14222#issuecomment-546057178.

Reply:
(Haven't looked at the new approach yet.) My only concern with leaving the `json.Number` values in the event is that conditions in processors probably won't work. For example the `range` condition only works on primitive values.